library/cc/stream_callbacks.cc

```diff
@@ -87,6 +87,17 @@ void* c_on_cancel(envoy_stream_intel, void* context) {
   return nullptr;
 }
 
+void* c_on_can_send_data(envoy_stream_intel, void* context) {
```

Comment:
Should this contain some indication of how much data can be sent on the next call?

Reply:
If it did, it could probably plug directly into some of the stream APIs on the platform stacks.

Reply:
Or is the idea not so much to avoid queueing/buffering as to simply inform when whatever was previously queued/buffered has been sent?

Reply:
Okay, reading the comments here, I think I understand better. It looks like the underlying API doesn't really support passing through any notion of available send window.